js/pushy.js

```diff
@@ -69,25 +69,44 @@
 		//hide submenu by default
 		$(submenuClass).addClass(submenuClosedClass);
 
-		$(submenuClass).on('click', function(){
+		$(submenuClass).on('click', function(e){
 	        var selected = $(this);
+	        var isNestedSubmenu = selected.parents(submenuClass).length;
 
 	        if( selected.hasClass(submenuClosedClass) ) {
 	            //hide opened submenus
 	            $(submenuClass).addClass(submenuClosedClass).removeClass(submenuOpenClass);
+
+	            // keep parent of nested submenu open if applicable
+	            if (isNestedSubmenu) {
+	                selected.parents(submenuClass)
+	                    .removeClass(submenuClosedClass)
+	                    .addClass(submenuOpenClass);
+							}
+
 	            //show submenu
 	            selected.removeClass(submenuClosedClass).addClass(submenuOpenClass);
 	        }else{
+	            // close parent of nested submenu if applicable
+	            if (isNestedSubmenu) {
+	                selected.parents(submenuClass)
+	                    .addClass(submenuClosedClass)
+	                    .removeClass(submenuOpenClass);
+	            }
+
 	            //hide submenu
 	            selected.addClass(submenuClosedClass).removeClass(submenuOpenClass);
 	        }
+
+	        e.stopPropagation();
```

Comment:
Great question @erikjung!

If we refer to [this markup](https://github.com/christophery/pushy/issues/90#issue-155380516), we will notice when we want nested submenus, you have a `.pushy-submenu` within another `.pushy-submenu`.

When the `stopPropagation()` is not included and you attempt to `click` to open a nested `.pushy-submenu`, it will properly _open_ the nested submenu, but since the event bubbles up, it will eventually get to the first-level `.pushy-submenu` which will actually _close_ the first-level `.pushy-submenu` not allowing you to view the open _nested_ `.pushy-submenu`.

I too wondered if I needed to truly stop the event propagation, but based on how the logic is currently set up, it seemed to be the only way I could get it to work.